### PR TITLE
Add loss, save_inference_model and logger

### PR DESCRIPTION
**Pull request by LielinJiang**
1. move Loss from model.py to loss.py
2. add save_inference_model
3. add logger
4. add unittest for multiple gpus train,eval and predict; transform; datasets; models;loss;logger.
5. add some comments
6. fix bug

### CODE REVIEW

File: hapi/logger.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    logger.propagate = False
+
+    format_str = '%(asctime)s - %(name)s - %(levelname)s - %(message)s'
+    logging.basicConfig(format=format_str, level=log_level)
+
+    # stdout logging: only local rank==0
+    local_rank = ParallelEnv().local_rank
+    if local_rank == 0:
+        ch = logging.StreamHandler(stream=sys.stdout)
+        ch.setLevel(logging.DEBUG)
```
Comment (heavengate) [inline hapi/logger.py:47]: should be set as `log_level` ?

File: hapi/loss.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
+# Copyright (c) 2019 PaddlePaddle Authors. All Rights Reserved.
```
Comment (heavengate) [inline hapi/loss.py:1]: 2019 -> 2020

File: hapi/loss.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+import os
+
+from paddle import fluid
+from paddle.fluid.framework import in_dygraph_mode, Variable
+from paddle.fluid.dygraph.base import to_variable
+
+__all__ = ['Loss', 'CrossEntropy', 'SoftmaxWithCrossEntropy']
+
+
+def to_list(value):
```
Comment (heavengate) [inline hapi/loss.py:26]: move these common functions to `utils.py` ?

File: hapi/test/dist_mnist.py [hunk 4]
```diff
@@ -0,0 +1,179 @@
+# Copyright (c) 2020 PaddlePaddle Authors. All Rights Reserved.
```
Comment (heavengate) [inline hapi/test/dist_mnist.py:1]: move tests directory to under `hapi/` will make whl contains test scripts, do we mean this?

File: hapi/model.py [hunk 5]
```diff
@@ -1208,6 +1169,59 @@ def predict(self,
...
+
+        Args:
+            dirname(str): The directory path to save the inference model.
+            model_filename(str|None): The name of file to save the inference program
+                                        itself. If is set None, a default filename
+                                        :code:`__model__` will be used.
+            params_filename(str|None): The name of file to save all related parameters.
+                                            If it is set None, parameters will be saved
+                                            in separate files .
+            export_for_deployment(bool): If True, programs are modified to only support
```
Comment (qingqing01) [inline hapi/model.py:1189]: 下面调用`fluid.io.save_inference_model`时始终传入的 infer_prog，而export_for_deployment=False时描述是为了训练。 这里觉得可以先把export_for_deployment参数去掉

File: hapi/test/dist_mnist.py [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+            global_pooling=global_pooling,
+            use_cudnn=use_cudnn)
+
+    def forward(self, inputs):
+        x = self._conv2d(inputs)
+        x = self._pool2d(x)
+        return x
+
+
+class MNIST(Model):
```
Comment (qingqing01) [inline hapi/test/dist_mnist.py:83]: 内部已经封装了LeNet了，觉得可以用，这里就不用写这些了

File: hapi/logger.py (hunk 1 above)
Comment (LielinJiang) [reply hapi/logger.py]: done, thanks

File: hapi/loss.py (hunk 2 above)
Comment (LielinJiang) [reply hapi/loss.py]: done, thanks

File: hapi/loss.py (hunk 3 above)
Comment (LielinJiang) [reply hapi/loss.py]: done, thanks

File: hapi/test/dist_mnist.py (hunk 4 above)
Comment (LielinJiang) [reply hapi/test/dist_mnist.py:1]: 看了下incubate目录，fleet下也有个test文件夹。

File: hapi/model.py (hunk 5 above)
Comment (LielinJiang) [reply hapi/model.py]: done，thanks

File: hapi/test/dist_mnist.py (hunk 6 above)
Comment (LielinJiang) [reply hapi/test/dist_mnist.py]: done, thanks

File: hapi/logger.py [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+import os
+import sys
+import logging
+import functools
+
+from paddle.fluid.dygraph.parallel import ParallelEnv
+
+
+@functools.lru_cache()
```
[the remark this answers is not in the document]
Comment (LielinJiang) [reply hapi/logger.py]: removed, thanks.

File: hapi/logger.py [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+        name (str): the root module name of this logger. Default: 'hapi'.
+        log_level (enum): log level. eg.'INFO', 'DEBUG', 'ERROR'. Default: logging.INFO.
+    Returns:
+        logging.Logger: a logger
+    """
+    logger = logging.getLogger(name)
+    logger.propagate = False
+
+    format_str = '%(asctime)s - %(name)s - %(levelname)s - %(message)s'
+    logging.basicConfig(format=format_str, level=log_level)
```
[the remark this answers is not in the document]
Comment (LielinJiang) [reply hapi/logger.py]: done, thanks.

**Pull request merged**